include/ck/utility/generic_memory_space_atomic_max.hpp

```diff
@@ -0,0 +1,44 @@
+#pragma once
+#include "data_type.hpp"
+
+namespace ck {
+
+template <typename X>
+__device__ X atomic_max(X* p_dst, const X& x);
```

Comment:
You can condense `int32_t`, `uint32_t`, and `float` specializations by implementing the function template instead of declaring it. For any other special cases like `float2_t` you can continue using template specializations.

Reply:
Fix in https://github.com/ROCmSoftwarePlatform/composable_kernel/pull/209/commits/d43efc6398986d4646d81674d9f354c64148248d

Reply:
As @asroy pointed out, this is done intentionally. Closing this conversation.